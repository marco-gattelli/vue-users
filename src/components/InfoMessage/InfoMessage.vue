<template>
  <div
    class="rounded p-4 border flex items-center justify-center"
    :class="{
      'bg-orange-300': variant === 'warning',
      'bg-red-300': variant === 'error',
    }"
    :data-testid="dataTestId"
  >
    {{ message }}
  </div>
</template>

<script setup lang="ts">
type Props = {
  message: string;
  dataTestId: string;
  variant: "warning" | "error";
};
const { message, variant, dataTestId } = defineProps<Props>();
</script>
