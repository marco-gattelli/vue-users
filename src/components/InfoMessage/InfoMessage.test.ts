import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import InfoMessage from "@/components/InfoMessage/InfoMessage.vue";

describe("InfoMessage", () => {
  it("should render without crash", function () {
    // Arrange.
    // Act.
    const { element } = mount(InfoMessage, {
      props: {
        message: "bar",
        variant: "warning",
      },
    });

    // Assert.
    expect(element).toMatchSnapshot();
  });

  it("should render error message", function () {
    // Arrange.
    // Act.
    const { element } = mount(InfoMessage, {
      props: {
        message: "baz",
        variant: "error",
      },
    });

    // Assert.
    expect(element).toMatchSnapshot();
  });
});
