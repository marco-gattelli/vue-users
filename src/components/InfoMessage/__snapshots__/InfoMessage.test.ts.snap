// Vitest Snapshot v1

exports[`InfoMessage > should render error message 1`] = `
<div
  class="rounded p-4 border flex items-center justify-center bg-red-300"
>
  baz
</div>
`;

exports[`InfoMessage > should render without crash 1`] = `
<div
  class="rounded p-4 border flex items-center justify-center bg-orange-300"
>
  bar
</div>
`;
